sklearn/model_selection/tests/test_split.py

```diff
@@ -0,0 +1,717 @@
+    labels = np.array([1, 2, 3, 4])
+    loo = LeaveOneOut()
+    lpo = LeavePOut(2)
+    kf = KFold(2)
+    skf = StratifiedKFold(2)
+    lolo = LeaveOneLabelOut()
+    lopo = LeavePLabelOut(2)
+    ss = ShuffleSplit()
+    ps = PredefinedSplit()
+    for i, cv in enumerate([loo, lpo, kf, skf, lolo, lopo, ss, ps]):
+        print(cv)
+        for train, test in cv.split(X, y, labels):
+            assert_not_equal(np.asarray(train).dtype.kind, 'b')
+            assert_not_equal(np.asarray(train).dtype.kind, 'b')
+
+
+def check_valid_split(train, test, n_samples=None):
+    # Use python sets to get more informative assertion failure messages
+    train, test = set(train), set(test)
+
+    # Train and test split should not overlap
+    assert_equal(train.intersection(test), set())
+
+    if n_samples is not None:
+        # Check that the union of train an test split cover all the indices
+        assert_equal(train.union(test), set(range(n_samples)))
+
+
+def check_cv_coverage(cv, X, y, labels, expected_n_iter=None, n_samples=None):
+    # Check that a all the samples appear at least once in a test fold
+    if expected_n_iter is not None:
+        assert_equal(cv.n_splits(X, y, labels), expected_n_iter)
+    else:
+        expected_n_iter = cv.n_splits(X, y, labels)
+
+    collected_test_samples = set()
+    iterations = 0
+    for train, test in cv.split(X, y, labels):
+        check_valid_split(train, test, n_samples=n_samples)
+        iterations += 1
+        collected_test_samples.update(test)
+
+    # Check that the accumulated test samples cover the whole dataset
+    assert_equal(iterations, expected_n_iter)
+    if n_samples is not None:
+        assert_equal(collected_test_samples, set(range(n_samples)))
+
+
+def test_kfold_valueerrors():
+    X1 = np.array([[1, 2], [3, 4], [5, 6]])
+    X2 = np.array([[1, 2], [3, 4], [5, 6], [7, 8], [9, 10]])
+    # Check that errors are raised if there is not enough samples
+    assert_raises(ValueError, next, KFold(4).split(X1))
+
+    # Check that a warning is raised if the least populated class has too few
+    # members.
+    y = np.array([3, 3, -1, -1, 2])
+
+    skf_3 = StratifiedKFold(3)
+    assert_warns_message(Warning, "The least populated class",
+                         next, skf_3.split(X2, y))
+
+    # Check that despite the warning the folds are still computed even
+    # though all the classes are not necessarily represented at on each
+    # side of the split at each split
+    with warnings.catch_warnings():
+        check_cv_coverage(skf_3, X2, y, labels=None, expected_n_iter=3,
+                          n_samples=len(y))
+
+    # Error when number of folds is <= 1
+    assert_raises(ValueError, KFold, 0)
+    assert_raises(ValueError, KFold, 1)
+    assert_raises(ValueError, StratifiedKFold, 0)
+    assert_raises(ValueError, StratifiedKFold, 1)
+
+    # When n_folds is not integer:
+    assert_raises(ValueError, KFold, 1.5)
+    assert_raises(ValueError, StratifiedKFold, 1.5)
```

Comment:
I suggest adding a test here with 2.0 instead of 1.5 and making sure that fails too if we agree on this.